components/sync15/src/sync.rs

```diff
@@ -36,38 +37,72 @@ pub trait Store {
     /// to handle "backfills" etc
     fn get_collection_request(&self) -> Result<CollectionRequest, failure::Error>;
 
-    fn reset(&self) -> Result<(), failure::Error>;
+    /// Get persisted sync IDs. If they don't match the global state we'll be
+    /// `reset()` with the new IDs.
+    fn get_sync_ids(&self) -> Result<(Option<CollSyncIds>), failure::Error>;
+
+    /// Reset the store without wiping local data, ready for a "first sync".
+    /// If called with Some(), then it means Sync is going to remain connected
+    /// so the specified guids should be persisted as part of resetting the state.
+    /// If called with None, it means Sync is being disconnected (although it
```

Comment:
While `None`/`Some(CollSyncIds)` is an accurate model, I do wonder if there's a better way to do this that won't lead to us forgetting the semantics around `reset()` 100 more times.

Maybe
```
enum ResetAction {
    Disconnected,
    NewSyncIds(CollSyncIds),
}
```
or something

Reply:
heh - fair enough. So I guess if `reset` takes this new enum, the "getter" might as well return one too. So I added a `StoreSyncAssoc`, which `reset` takes. I also renamed `get_sync_ids()` to `get_sync_assoc()` and it returns one of these too. The next step is probably to rename `reset()` to `set_sync_assoc()`, but I didn't do that.